### DISCUSSION TITLE
Copilot.vim in neovim not display suggestion in virtual text.

### DISCUSSION THREAD

**Discussion by nanozuki** [community/community#64834] [Copilot Conversations] [answered]
### Select Topic Area

Product Feedback

### Body

When I use copilot.nvim 1.10.0 and 1.10.1 （last two commits）, the suggestion of copilot won't displayed in virtual text.
My config is (use lazy.nvim): 

```lua
  {
    'github/copilot.vim',
    init = function()
      vim.g.copilot_no_tab_map = true
    end,
    config = function()
      vim.keymap.set('i', '<C-e>', [[copilot#Accept("\<CR>")]], {
        silent = true,
        expr = true,
        script = true,
        replace_keycodes = false,
      })
    end,
  },
```

neovim version:

```
NVIM v0.9.1
Build type: Release
LuaJIT 2.1.0-beta3

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "
/nix/store/qjsdjkbcwrrbnjz8n4in8z5vd37nmznd-neovim-unwrapped-0.9.1/share/nvim
"
```

`:Copilot status`:

```
Copilot: Enabled and online
```

**Comment by arafajs** [comment]
Check your Neovim configuration file for any settings that may be interfering with Copilot. For example, the `copilot_no_virtual_text` setting will disable virtual text suggestions.

**Reply by nanozuki** [reply]
There is no "copilot_no_virtual_text" in copilot.vim and my own config. Do you read what I wrote and the code of copilot.vim?

**Comment by omercsp** [comment]
This happens to me as well. I use copilot in vim, but the problem is the same: no suggestions are displayed. 

I looked into it a bit:

1. It has something to do the with the `copilot_no_tab_map` setting. As with @nanozuki, I have mine set to `v:true` as well (`let g:copilot_no_tab_map = v:true` in my case). Once this line is removed, at least in my environment, suggestion appear as normal. This isn't a solution of course, as I don't wont copilot to overwrite my mapping.
2. I drilled down a bit further, and the following line is to be blamed (plugin/copilot.vim, function `s:MapTab` in v1.10.1):
```
exe 'imap <script><silent><nowait><expr> <Tab> copilot#Accept(' . tab_fallback . ')'
```

Assuming the `copilot_no_tab_map` isn't set to `v:true`, commenting out the mentioned line prevents suggestion to appear in insert mode. This is weird; to my understanding, all this line does is mapping the `tab` key, but still, this is where we are at.

**Comment by omercsp** [comment]
More info, and possible solution (for me at least): 
My copilot plug-in was installed as an optional plugin (i.e. in and `opt` directory), and was loaded explicitly using `packadd`, where the map variable was set before the plugin was loaded. Moving to a `start` folder instead of an `opt` (removing the `packadd` command), seems to solve this issue. 

For me, for now, it's OK, but this doesn't strike me as the desired behaviour.

**Comment by tpope** [comment, marked as answer]
This should be fixed in Copilot.vim 1.10.2. Sorry about that!
